### PR TITLE
Add no results message 

### LINKED ISSUE #42: Add empty state in search
Add some messaging/empty stage message when no results:

image

### PR DESCRIPTION
**Pull request by andesol**
## Done 

Add a short message when the search does not return any spec

## QA

- Open demo (or locally: `dotrun`)
- Type some random words in the search bar until no spec is found
- Verify there is a message saying "no specs found"

## Issue
Fixes #42

### CODE REVIEW

File: src/styles.scss [hunk 1]
```diff
@@ -5,6 +5,10 @@ $increase-font-size-on-larger-screens: false;
 @include vanilla;
 @include vf-p-icon-comments;
 
+main {
+  padding-bottom: $spv--strip-regular;
```
Comment (andesol) [inline src/styles.scss:9]: No related with this PR, but there was no space between the cards and the footer

File: templates/_base.html [hunk 2]
```diff
@@ -105,6 +111,18 @@
...
+        if (count === 0 && !message) {
+            cardsGrid.classList.add("u-hide")
+            message = document.getElementById("no-results").content.cloneNode(true);
+            main.appendChild(message)
+        } else if (count !== 0 && message) {
+          cardsGrid.classList.remove("u-hide");
+          main.removeChild(message);
+        }
+      }
+
```
Comment (goulinkh) [inline templates/_base.html]: I think you don't have do DOM update with the template pattern, I think it's simpler to do the same as how we handle spec card elements, where the empty state message is always in the DOM and you only toggle the class `u-hide`, to make the project consistent, as we are not using templates anywhere else in the project and there is only one element created out of the template.

File: templates/index.html [hunk 3]
```diff
@@ -44,9 +44,16 @@
...
 <div class="l-fluid-breakout__main" id="cards">
   {% for spec in specs %} {% with spec=spec %} {% include "_card.html" %} {%
   endwith %} {% endfor %}
 </div>
+<template id="no-results">
+  <div
+    class="l-fluid-breakout__main u-align-text--center"
+    id="no-results-message"
+  >
+    <strong>No specs found</strong>
```
Comment (goulinkh) [inline templates/index.html]: this should be an `h2` or `h3`

Comment (andesol) [reply templates/index.html]: :+1:

File: templates/_base.html (hunk 2 above)
Comment (andesol) [reply templates/_base.html]: Good point. This was my first intuition, but there were some particularities with the markup:

1. We were using card's parent for sorting purposes, so this wouldn't work straightaway:
```html
<div class="l-fluid-breakout__main" id="cards">
      <h2 id="no-results" class="u-hide">No specs found</h2>
      <div>...</div>
      <div>...</div>
      <div>...</div>
   </div>
</div>
```

2. `l-fluid-breakout__main` expect all the components to be at the same level, so this wouldn't work either:
```html
<div class="l-fluid-breakout__main">
  <h2 id="no-results" class="u-hide">No specs found</h2>
  <div id='cards'>
      <div>...</div>
      <div>...</div>
      <div>...</div>
   </div>
</div>
```

But giving it a second thought, it is true that changing a bit the sort function and playing with the css solves the issue and indeed simplifies things. Thanks!

File: templates/index.html [hunk 4]
```diff
@@ -44,9 +44,17 @@
 </div>
 
 <div class="l-fluid-breakout__aside">{% include "_filters.html" %}</div>
-
 <div class="l-fluid-breakout__main" id="cards">
+  <h2 id="no-results" class="u-hide u-align-text--center">No specs found</h2>
   {% for spec in specs %} {% with spec=spec %} {% include "_card.html" %} {%
   endwith %} {% endfor %}
 </div>
+<template id="no-results">
```
Comment (goulinkh) [inline templates/index.html]: do you still need the `<template>` tag here?

Comment (andesol) [reply templates/index.html]: No, nice catch!

**Review by goulinkh — APPROVED** [review]
LGTM

**Pull request merged**